src/basic/unit-name.c

```diff
@@ -507,6 +509,42 @@ int unit_name_template(const char *f, char **ret) {
         return 0;
 }
 
+int unit_name_hash_long(const char *name, char **ret) {
+        _cleanup_free_ char *s = NULL;
+        char hash[17] = {}, n[UNIT_NAME_MAX] = {}, *suffix;
+        uint64_t h;
+        uint8_t k[16] = {};
+        int i;
+
+        if (strlen(name) < UNIT_NAME_MAX)
+                return 0;
```

Comment:
as per coding style, our functions should initialize all return values on success. here you return success, and thus should initialize `ret`, probably just to `NULL`